runtime/src/main/java/org/corfudb/runtime/view/replication/ChainReplicationProtocol.java

```diff
@@ -143,13 +143,14 @@ public ILogData peek(RuntimeLayout runtimeLayout, long globalAddress) {
      * @param data          The data to propagate, or NULL,
      *                      if it is to be a hole.
      */
-    protected void propagate(RuntimeLayout runtimeLayout,
+    private void propagate(RuntimeLayout runtimeLayout,
                              long globalAddress,
                              @Nullable ILogData data) {
         int numUnits = runtimeLayout.getLayout().getSegmentLength(globalAddress);
 
         for (int i = 1; i < numUnits; i++) {
-            log.info("Propogate[{}]: chain {}/{}", Token.of(runtimeLayout.getLayout().getEpoch(), globalAddress),
+            log.trace("Propogate[{}]: chain {}/{}", Token.of(runtimeLayout.getLayout().getEpoch(),
```

Comment:
Now that we're on this, can you please correct the typo in 'propagate' here and also in the catch clause..

Reply:
Done